core/sr-primitives/src/sr_arithmetic.rs

```diff
@@ -0,0 +1,1013 @@
+					$name::from_percent(90)
+				);
+				assert_eq!(
+					$name::from_percent(50).saturating_add($name::from_percent(50)),
+					$name::from_percent(100)
+				);
+				assert_eq!(
+					$name::from_percent(60).saturating_add($name::from_percent(50)),
+					$name::from_percent(100)
+				);
+
+				assert_eq!(
+					$name::from_percent(60).saturating_sub($name::from_percent(50)),
+					$name::from_percent(10)
+				);
+				assert_eq!(
+					$name::from_percent(60).saturating_sub($name::from_percent(60)),
+					$name::from_percent(0)
+				);
+				assert_eq!(
+					$name::from_percent(60).saturating_sub($name::from_percent(70)),
+					$name::from_percent(0)
+				);
+
+				assert_eq!(
+					$name::from_percent(50).saturating_mul($name::from_percent(50)),
+					$name::from_percent(25)
+				);
+				assert_eq!(
+					$name::from_percent(20).saturating_mul($name::from_percent(20)),
+					$name::from_percent(4)
+				);
+				assert_eq!(
+					$name::from_percent(10).saturating_mul($name::from_percent(10)),
+					$name::from_percent(1)
+				);
+			}
+
+			#[test]
+			fn per_thing_square_works() {
+				assert_eq!($name::from_percent(100).square(), $name::from_percent(100));
+				assert_eq!($name::from_percent(50).square(), $name::from_percent(25));
+				assert_eq!($name::from_percent(10).square(), $name::from_percent(1));
+				assert_eq!($name::from_percent(2).square(), $name::from_parts(4 * $max / 10_000));
+			}
+
+			#[test]
+			fn per_things_div_works() {
+				// normal
+				assert_eq!($name::from_percent(10) / $name::from_percent(20),
+					$name::from_percent(50)
+				);
+				assert_eq!($name::from_percent(10) / $name::from_percent(10),
+					$name::from_percent(100)
+				);
+				assert_eq!($name::from_percent(10) / $name::from_percent(0),
+					$name::from_percent(100)
+				);
+
+				// will not overflow
+				assert_eq!($name::from_percent(10) / $name::from_percent(5),
+					$name::from_percent(100)
+				);
+				assert_eq!($name::from_percent(100) / $name::from_percent(50),
+					$name::from_percent(100)
+				);
+			}
+		}
+	};
+}
+
+implement_per_thing!(
+	Permill,
+	test_permill,
+	[u32, u64, u128],
+	1_000_000u32,
+	u32,
+	u64,
+	"_Parts per Million_",
+);
+implement_per_thing!(
+	Perbill,
+	test_perbill,
+	[u32, u64, u128],
+	1_000_000_000u32,
+	u32,
+	u64,
+	"_Parts per Billion_",
+);
+implement_per_thing!(
+	Perquintill,
+	test_perquintill,
+	[u64, u128],
+	1_000_000_000_000_000_000u64,
+	u64,
+	u128,
+	"_Parts per Quintillion_",
+);
+implement_per_thing!(
+	Percent,
+	test_per_cent,
+	[u32, u64, u128],
+	100u32,
+	u32,
+	u32,
+	"_Percent_",
+);
+
+/// An unsigned fixed point number. Can hold any value in the range [-9_223_372_036, 9_223_372_036]
+/// with fixed point accuracy of one billion.
+#[cfg_attr(feature = "std", derive(Debug))]
+#[derive(Encode, Decode, Default, Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct Fixed64(i64);
+
+/// The maximum value of the `Fixed64` type
```

Comment:
what does that mean doesn't Fixed64 can be more than a billion actually ?

Reply:
bad doc. updated.